lasy/utils/grid.py

```diff
@@ -88,7 +88,8 @@ def set_is_envelope(self, is_envelope):
             self.dtype = "complex128"
         else:
             self.dtype = "float64"
-        self.temporal_field = self.temporal_field.astype(dtype=self.dtype)
+        if hasattr(self, "temporal_field"):
+            self.temporal_field = self.temporal_field.astype(dtype=self.dtype)
```

Comment:
This avoids unnecessary casting from complex to float at initialization with `is_envelope = False`

Reply:
Indeed, thanks for the comment!